### PR TITLE
Depend on attrs ≥19.2.0; use eq instead of cmp

### PR DESCRIPTION
**Pull request by twm**
Set the minimal dependency to attrs 19.2.0, which provides `eq` and deprecates `cmp`. 
See the [attrs changelog](https://www.attrs.org/en/stable/changelog.html#id2).

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/9710
* [x] The changes pass minimal style checks (see: https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted )
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles )
* [x] I have updated the automated tests. (N/A)
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.

### CODE REVIEW

File: src/twisted/internet/address.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ class HostnameAddress(object):
 
 
 
-@attr.s(hash=False, repr=False, cmp=False)
+@attr.s(hash=False, repr=False, eq=False)
```
Comment (moshez) [inline src/twisted/internet/address.py:101]: I was wondering what `order` was set to. I see the default is "same as eq", so it is doing the right thing, but I wonder if it's worthwhile calling it out explicitly.

**Review by moshez — APPROVED** [review]
Thank you for working on this! Always nice to be ahead of deprecations!

One minor non-blocking question, otherwise seems good to me.

File: src/twisted/internet/address.py (hunk 1 above)
Comment (twm) [reply src/twisted/internet/address.py:101]: The attrs defaults are perfectly sensible now that `cmp` has been split, so I don't think that's necessary.

**Comment by twm** [comment]
Thank you for the review, @moshez! Landing now.

**Pull request merged**